### PR TITLE
[IMP][11.0][web_widget_color] Refactor Module

### PR DESCRIPTION
**Pull request by Tardo**
- Fix issue #841 
- Follow new readme structure
- Cleaned unused CSS code & Use Less
- Improved JS Code
- Improved XML field
- Fix style to looks like readme images:
ListView
- Updated jscolor library to last version

### CODE REVIEW

**Review by etobella — APPROVED** [review]
LGTM :+1:

File: web_widget_color/static/src/js/widget.js [hunk 1]
```diff
@@ -1,53 +1,49 @@
...
-                this.$(".oe_form_char_content").text(show_value);
-                this.$('span').css("background-color", show_value);
-            } else {
-                this.$(".oe_form_char_content").text(show_value);
-                this.$('span').css("background-color", show_value);
 
-            }
+        _renderEdit: function() {
+            var self = this;
+            // FIXME: For some reason needs 1ms delay for initilize jscolor...
```
Comment (simahawk) [inline web_widget_color/static/src/js/widget.js]: call super and hook to returned deferred w/ "then"?
Also, is probably better to initialize the widget like `var myPicker = new jscolor.color($('#myField'), {})`

File: web_widget_color/static/src/js/widget.js [hunk 2]
```diff
@@ -1,53 +1,49 @@
...
+                _.each(recordWidgets, function (widget) {
+                    var $el = widget.getFocusableElement();
+                    if ($el.hasClass('jscolor') && $el.is(":focus")) {
+                        canUnselect = false;
+                        return false;
+                    }
+                });
+            }
+
+            return canUnselect?this._super.apply(this, arguments):$.when();
```
Comment (simahawk) [inline web_widget_color/static/src/js/widget.js]: maybe a normal IF/ELSE block would make this more readable?

**Review by simahawk — CHANGES_REQUESTED** [review]
well done :)

File: web_widget_color/static/src/js/widget.js (hunk 2 above)
Comment (tarteo) [reply web_widget_color/static/src/js/widget.js]: Can you instead of $.when() return a $.Deferred().resolve()?

**Review by tarteo — CHANGES_REQUESTED** [review]

**Comment by Tardo** [comment]
Thanks for reports... hope that now works properly :)

File: web_widget_color/static/src/js/widget.js (hunk 1 above)
Comment (Tardo) [reply web_widget_color/static/src/js/widget.js]: @simahawk Odoo documentation says that renderEdit return Deferred/Undefined.. but 'InputField' renderEdit return an JQueryObject/HTMLElement... :S

**Comment by Tardo** [comment]
Added option 'readonly_mode':

```<field name="the_field" widget="color" options="{'readonly_mode':'text'}" />```

- readonly_mode = 'text'
color_ot

- readonly_mode = 'color'
color_oc

- readonly_mode undefined or other value:
color_n

**Review by simahawk — APPROVED** [review]
LGTM, you could squash some commits maybe

**Comment by pedrobaeza** [comment]
We can squash on merge. @tarteo do you confirm everything is good on your side?

**Comment by pedrobaeza** [comment]
@tarteo?

**Review by tarteo — APPROVED** [review]

**Pull request merged**